addons/sourcemod/scripting/szf/pickupweapons.sp

```diff
@@ -500,10 +444,10 @@ public void PickupWeapon(int iClient, int iIndex, int iTarget)
 	}
 	
 	// generate and equip weapon
-	int iWeapon = TF2_CreateAndEquipWeapon(iClient, iIndex);
+	int iWeapon = TF2_CreateAndEquipWeapon_eWeapon(iClient, wep);
```

Comment:
Not a fan of having `TF2_CreateAndEquipWeapon_eWeapon` when it basically the same as `TF2_CreateAndEquipWeapon` but with extra stuffs, could've use original with extra stuffs here.

(It shouldn't matter which order applying attribute and `DispatchSpawn` goes first)